Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp

```diff
@@ -268,4 +243,31 @@ namespace ROS2::VehicleDynamics::Utilities
         }
         return AZStd::clamp(commandVelocity, -maxVelocity, maxVelocity);
     }
+
+    void SetWheelRotationSpeed(const  VehicleDynamics::WheelDynamicsData& data, float wheelRotationSpeed)
+    {
+        if (data.m_isArticulation)
+        {
+            PhysX::ArticulationJointRequestBus::Event(
+                data.m_wheelEntity, &PhysX::ArticulationJointRequests::SetDriveTargetVelocity, data.m_axis, wheelRotationSpeed);
+        }
+        else
+        {
+            PhysX::JointRequestBus::Event( AZ::EntityComponentIdPair(data.m_wheelEntity,data.m_wheelJoint), &PhysX::JointRequests::SetVelocity, wheelRotationSpeed);
+        }
+    }
+
+    AZ::Transform GetJointTransform(const VehicleDynamics::WheelDynamicsData& data)
+    {
+        AZ::Transform hingeTransform{ AZ::Transform::Identity() };
+        if (data.m_isArticulation)
+        {
+            hingeTransform = AZ::Transform::Identity();
+        }
+        else
+        {
+            PhysX::JointRequestBus::EventResult(hingeTransform, AZ::EntityComponentIdPair(data.m_wheelEntity,data.m_wheelJoint), &PhysX::JointRequests::GetTransform);
+        }
```

Comment:
The if block isn't changing the hingeTransform value at all.
The condition can be converted to check if articulation is not enabled.
```suggestion
        if (!data.m_isArticulation)
        {
            PhysX::JointRequestBus::EventResult(hingeTransform, AZ::EntityComponentIdPair(data.m_wheelEntity,data.m_wheelJoint), &PhysX::JointRequests::GetTransform);
        }
```